lib/postgres.dart

```diff
@@ -438,3 +442,102 @@ enum QueryMode {
   /// Simple Query Protocol
   simple,
 }
+
+/// The isolation level of a transaction determines what data the transaction
+/// can see when other transactions are running concurrently.
+enum IsolationLevel {
+  /// A statement can only see rows committed before it began.
+  /// This is the default.
+  readCommitted('READ COMMITTED'),
+
+  /// All statements of the current transaction can only see rows committed
+  /// before the first query or data-modification statement was executed in
+  /// this transaction.
+  repeatableRead('REPEATABLE READ'),
+
+  /// All statements of the current transaction can only see rows committed
+  /// before the first query or data-modification statement was executed in
+  /// this transaction. If a pattern of reads and writes among concurrent
+  /// serializable transactions would create a situation which could not have
+  /// occurred for any serial (one-at-a-time) execution of those transactions,
+  /// one of them will be rolled back with a serialization_failure error.
+  serializable('SERIALIZABLE'),
+
+  /// One transaction may see uncommitted changes made by some other transaction.
+  /// In PostgreSQL READ UNCOMMITTED is treated as READ COMMITTED.
+  readUncommitted('READ UNCOMMITTED'),
+  ;
+
+  /// The SQL identifier of the isolation level including "ISOLATION LEVEL" prefix.
+  final String _sqlName;
+
+  const IsolationLevel(String identifier)
+      : _sqlName = 'ISOLATION LEVEL $identifier';
+}
+
+/// The transaction access mode determines whether the transaction is read/write
+/// or read-only.
+enum AccessMode {
```

Comment:
Given that there are only two possible values, maybe it should also be a `bool? readOnly` property on `TransactionMode`, similar to `deferrable`?

Reply:
I'm unsure, but we could make the deferrable into an enum too... it is nice to have the documentation on the fields.